WireDomain/Sources/WireDomain/Repositories/Conversations/ConversationLocalStore.swift

```diff
@@ -461,6 +481,54 @@ public final class ConversationLocalStore: ConversationLocalStoreProtocol {
             )
         }
     }
+    
+    public func updateTypingUsers(
+        conversationID: NSManagedObjectID,
+        usersID: Set<NSManagedObjectID>
+    ) async {
+        
+        // Switching to main context
+        let uiContext = await context.perform { [context] in
+            context.zm_userInterface
+        }
+        guard let uiContext else {
+            return
+        }
+        
+        uiContext.performGroupedBlock { [weak self] in
+            if let conversation = uiContext.object(with: conversationID) as? ZMConversation {
+                
+                let users = usersID.compactMap {
+                    uiContext.object(with: $0) as? ZMUser
+                }
+                
+                uiContext.typingUsers?.update(
+                    typingUsers: Set(users),
+                    in: conversation
+                )
```

Comment:
The underlying implementation was not accessible from WireDomain (it was part of WireSyncEngine), so TypingUsers object and the related NSManagedObjectContext extension were moved from WireSyncEngine to WireDataModel.

At the app level, the `ConversationInputBarViewController` relies on this code. Now that it is accessible from WireDataModel, we can reuse the same methods from WireDomain so when we plug our new quick sync mechanism, UI won't break (although it should probably be refactored as well eventually)

Reply:
This definitely needs refactoring... we shouldn't store anything on the managed object context. Do you think it's something we can do in this PR or it's too big?

Reply:
I agree but I think it's too big and sensitive to be done in this PR. It means revisiting some legacy code at the app level to introduce a new way of communication between WireDomain and the app layer (which we haven't fully figured out yet) while still making sure legacy code don't break. It could lead to introducing bugs, worse breaking the feature since it still relies a lot on WireSyncEngine. 

I feel like we're still not at the stage yet to start revisiting legacy app layer code so it fits in with our new sync mechanism. 

A workaround solution that came to mind was to get rid of these intermediary objects and operations and directly observe the notification from the UI component. (but again that means revisiting the app layer code)

What do you think ?

Reply:
also mentioning @netbe for the discussion

Reply:
Makes sense, let's create a ticket though to come back to this.

Reply:
@jullianm create the ticket and we can tackle that later